tests/v3/test_store/test_stateful_store.py

```diff
@@ -0,0 +1,190 @@
+# Stateful tests for arbitrary Zarr stores.
+
+import asyncio
+
+import hypothesis.strategies as st
+from hypothesis import note
+from hypothesis.stateful import (
+    RuleBasedStateMachine,
+    invariant,
+    precondition,
+    rule,
+)
+
+from zarr.abc.store import Store as StoreABC
+from zarr.buffer import Buffer, default_buffer_prototype
+from zarr.store import LocalStore, MemoryStore, RemoteStore
+
+# from strategies_store import StoreStatefulStrategies, key_ranges
+from zarr.testing.strategies import key_ranges, paths
+
+# zarr spec: https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html
+
+
+class SyncMemStoreWrapper(StoreABC):
+    def __init__(self, store_type):
+        """Store to hold async functions that map to StoreABC abstract methods"""
+        if store_type in ("Memory", "memory"):
+            self.store = MemoryStore(mode="w")  # store_container.store
+        elif store_type in ("Local", "local"):
+            self.store = LocalStore(mode="w")
+        elif store_type in ("Remote", "remote"):
+            self.store = RemoteStore(mode="w")
+        else:
+            raise ValueError(f"Invalid item_type: {store_type}")
+        # Unfortunately, hypothesis' stateful testing infra does not support asyncio
+
+    # So we redefine sync versions of the Store API.
+    # https://github.com/HypothesisWorks/hypothesis/issues/3712#issuecomment-1668999041
+    async def set(self, key, data_buffer):  # buffer is value
+        await self.store.set(key, data_buffer)
```

Comment:
```suggestion
    def set(self, key, data_buffer):  # buffer is value
        return asyncio.run(self.store.set(key, data_buffer))
```

and so on...

Reply:
I get an error like this when i try these `E           ValueError: a coroutine was expected, got <async_generator object MemoryStore.list at 0x7f90d0a3ff40>` 
I'm still trying to figure out async things but from what I can tell I think it's asycing for an `async def... await` instead of `asyncio.run()` does that sound right to you ?

Reply:
ah nevermind, I think i mixed that up